src/serializer/visitors.js

```diff
@@ -56,6 +56,27 @@ function replaceName(path, residualFunctionBinding, name, data) {
   }
 }
 
+function getLiteralTruthiness(node) {
+  // Returns { known: boolean, value?: boolean }. 'known' is true only if this is a literal of known truthiness and with no side effects; if 'known' is true, 'value' is its truthiness.
```

Comment:
No need to hide type info in comment, you can make it a real type annotation :-)

Reply:
To avoid object allocations, a simpler scheme of `void | false | true` would be sufficient.

Reply:
I generally figure the cost of this sort of object allocation is worth the additional readability, at least in modern engines. v8 at least is pretty good at optimizing the creation and use of this sort of object.